components/backup-stream/src/subscription_manager.rs

```diff
@@ -428,23 +346,22 @@ where
     ///
     /// a two-tuple, the first is the handle to the manager, the second is the
     /// operator loop future.
-    pub fn start<E, RT>(
-        initial_loader: InitialDataLoader<E, TlsCdcHandle<E, RT>>,
+    pub fn start<E, HInit, HChkLd>(
+        initial_loader: InitialDataLoader<E, HInit>,
         regions: R,
-        router: &RT,
         observer: BackupStreamObserver,
         meta_cli: MetadataClient<S>,
         pd_client: Arc<PDC>,
         scan_pool_size: usize,
-        resolver: BackupStreamResolver<RT, E>,
+        resolver: BackupStreamResolver<HChkLd, E>,
     ) -> (Self, future![()])
     where
         E: KvEngine,
-        RT: CdcHandle<E> + 'static,
+        HInit: CdcHandle<E> + Sync + 'static,
+        HChkLd: CdcHandle<E> + 'static,
```

Comment:
Can they be the same?

Reply:
In fact they are different type(`Arc<Mutex<RaftRouter>>` and `RaftRouter`) and used by different components(one for fetching the initial data and one for checking the leader). I think keep them different types will be fine for now.